<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>fscLVM documentation &#8212; fscLVM 1.0.0.dev1 documentation</title>
    
    <link rel="stylesheet" href="_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0.0.dev1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="fscLVM 1.0.0.dev1 documentation" href="#" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="nav-item nav-item-0"><a href="#">fscLVM 1.0.0.dev1 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="fsclvm-documentation">
<h1>fscLVM documentation<a class="headerlink" href="#fsclvm-documentation" title="Permalink to this headline">¶</a></h1>
<p>f-scLVM is a scalable modelling framework for single-cell RNA-seq data that can be used to dissect and model single-cell transcriptome heterogeneity, thereby allowing to identify biological drivers of cell-to-cell variability and model confounding factors.</p>
<p>Software by Florian Buettner and Oliver Stegle. f-scLVM is explained in detail in the accompanying publication [1].</p>
<p>[1] Buettner, F.,Pratanwanich, N., Marioni, J., Stegle, O. Scalable latent-factor models applied to single-cell RNA-seq data separate biological drivers from confounding effects. Submitted</p>
<div class="section" id="tutorial">
<h2>Tutorial<a class="headerlink" href="#tutorial" title="Permalink to this headline">¶</a></h2>
<p>All steps required to run f-scLVM are illustrated in a jupyter notebook that can be viewed <a class="reference external" href="http://nbviewer.jupyter.org/github/pmbio/f-scLVM/blob/master/py/demo/ipynb/f-scLVM.ipynb">interactively</a>. A brief tutorial on how to load data, fit the model and plot the results is summarised in the following secitons.</p>
<div class="section" id="input">
<h3>Input<a class="headerlink" href="#input" title="Permalink to this headline">¶</a></h3>
<p>f-scLVM expects a hdf file containing the normalised, log transformed gene expression data as well as a set of annotations. We provide an R script (LINK) that can be used to generate this input from a gene expression matrix, using annotation from REACTOME or MSigDB.</p>
<p>Usage is also illustrated in the an ipython notebook.</p>
<p>We provide a <code class="docutils literal"><span class="pre">load_hdf5</span></code> function for loading the input data form such hdf file.</p>
</div>
<div class="section" id="model-initialisation-and-fitting">
<h3>Model initialisation and fitting<a class="headerlink" href="#model-initialisation-and-fitting" title="Permalink to this headline">¶</a></h3>
<p>An f-scLVM model can be initialised using the <code class="docutils literal"><span class="pre">intiFA</span></code> convenience function. Arguments can be used to specify options, incuding number of unannotated factors (<cite>nHidden</cite>), minimum number of genes in a pathway (<cite>minGenes</cite>), whether to use the fast option by pruning genes (<cite>pruneGenes</cite>), noise model (<cite>noise</cite>) and the data directory (<cite>data_dir</cite>). Once a model is initialised, it can be fit using the <code class="docutils literal"><span class="pre">iterate</span></code> method.</p>
</div>
<div class="section" id="diagnostics-plotting-and-saving">
<h3>Diagnostics, plotting and saving.<a class="headerlink" href="#diagnostics-plotting-and-saving" title="Permalink to this headline">¶</a></h3>
<p>The <code class="docutils literal"><span class="pre">printDiagnostics</span></code> function can be used to print diagnositcs based ont eh number of genes included/excluded by the model.  If more than 100% of gene annotations are changed for at least one factor, the model should be re-fitted with sparse unannotated facotrs.</p>
</div>
</div>
<div class="section" id="loading-data-and-model-initialisation">
<h2>Loading data and model initialisation<a class="headerlink" href="#loading-data-and-model-initialisation" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="fscLVM.load_hdf5">
<code class="descclassname">fscLVM.</code><code class="descname">load_hdf5</code><span class="sig-paren">(</span><em>dFile</em>, <em>data_dir='../../../data/'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#load_hdf5"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.load_hdf5" title="Permalink to this definition">¶</a></dt>
<dd><p>Load input file for f-scLVM</p>
<p>Loads an hdf file and extracts all the inputs required by f-scLVM</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>dFile</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; String contaning the file name of the hdf file with the input data.</li>
<li><strong>data_dir</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; Name of the directory containing the input hdf5 file.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">An dictionary containing all the inputs required by f-scLVM.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fscLVM.initFA">
<code class="descclassname">fscLVM.</code><code class="descname">initFA</code><span class="sig-paren">(</span><em>Y</em>, <em>terms</em>, <em>I</em>, <em>nHidden=3</em>, <em>nHiddenSparse=0</em>, <em>pruneGenes=False</em>, <em>FPR=0.99</em>, <em>FNR=0.001</em>, <em>noise='gauss'</em>, <em>minGenes=20</em>, <em>do_preTrain=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#initFA"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.initFA" title="Permalink to this definition">¶</a></dt>
<dd><p>Initialise the f-scLVM factor analysis model.</p>
<p>Required 3 inputs are first, a gene expression matrix <cite>Y</cite> containing normalised count values of <cite>N</cite> cells and <cite>G</cite>
variable genes in log-space, second a vector <cite>terms</cite> contaning the names of all annotated gene set (correspondig to annotated factors)
and third, a binary inidcator matrix <cite>I</cite> linking <cite>G</cite> genes to <cite>K</cite> terms by indicating which genes are annotated to each factor.
A variety of options can be specified as described below.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>Y</strong> (<em>array_like</em>) &#8211; Matrix of normalised count values of <cite>N</cite> cells
and <cite>G</cite> variable genes in log-space.
Dimension (<span class="math">\(N\times G\)</span>).</li>
<li><strong>terms</strong> (<em>vector_like</em>) &#8211; Names of <cite>K</cite> annotated gene sets. Dimension
(<span class="math">\(K\times 0\)</span>).</li>
<li><strong>P_I</strong> (<em>array_like</em>) &#8211; Inidicator matirx specifying
whether a gene is annotated to a specific factor.
Dimension (<span class="math">\(G\times K\)</span>).</li>
<li><strong>FNR</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#float" title="(in Python v2.7)"><em>float</em></a>) &#8211; False negative rate of annotations.
Defaults to 0.001</li>
<li><strong>FPR</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#float" title="(in Python v2.7)"><em>float</em></a>) &#8211; False positive rate of annotations.
Defaults to 0.99</li>
<li><strong>nHidden</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Number of unannotated dense factors. Defaults to 3.</li>
<li><strong>nHiddenSparse</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Number of unannotated sparse factors. Defaults to 0.
This value should be changed to e.g. 5 if the diagnositcs fail.</li>
<li><strong>pruneGenes</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; prune genes that are not annotated to a least one factor. This option allows fast inference and
should be set to <cite>True</cite> either if the
key objective is to rank factors or if the annotations cover all genes of interest.
Defaults to <cite>False</cite>.</li>
<li><strong>noise</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; Specifies the observation noise model. Should be either <cite>&#8216;gauss&#8217;</cite>,`&#8217;hurdle&#8217;` or <cite>&#8216;poisson&#8217;</cite>.
Defaults to <cite>gauss</cite>.</li>
<li><strong>minGenes</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; minimum number of genes required per term to retain it
Defaults to <cite>20</cite>.</li>
<li><strong>do_preTrain</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Boolean switch indicating whether pre-training should be used to establish the initial
update order. Can be set to <cite>False</cite> for very large datasets.
Defaults to <cite>True</cite></li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A <code class="xref py py-class docutils literal"><span class="pre">fscLVM.CSparseFA</span></code> instance.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fscLVM.preTrain">
<code class="descclassname">fscLVM.</code><code class="descname">preTrain</code><span class="sig-paren">(</span><em>Y</em>, <em>terms</em>, <em>P_I</em>, <em>noise='gauss'</em>, <em>nFix=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#preTrain"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.preTrain" title="Permalink to this definition">¶</a></dt>
<dd><p>Pre-train the f-scLVM factor analysis model.</p>
<p>Helper function to pre-train the f-scLVM factor analysis model to achieve
faster convergence and obtain an initial update order. Called by <cite>initFA</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>Y</strong> (<em>array_like</em>) &#8211; Matrix of normalised count values of <cite>N</cite> cells
and <cite>G</cite> variable genes in log-space.
Dimension (<span class="math">\(N\times G\)</span>).</li>
<li><strong>terms</strong> (<em>vector_like</em>) &#8211; Names of <cite>K</cite> annotated gene sets. Dimension
(<span class="math">\(K\times 0\)</span>).</li>
<li><strong>P_I</strong> (<em>array_like</em>) &#8211; Matrix specifying the likelihood of
whether a gene is annotated to a specific factor.
Dimension (<span class="math">\(G\times K\)</span>).</li>
<li><strong>noise</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; Specifies the observation noise model. Should be either <cite>&#8216;gauss&#8217;</cite>,`&#8217;hurdle&#8217;` or <cite>&#8216;poisson&#8217;</cite>.
Defaults to <cite>gauss</cite>.</li>
<li><strong>nFix</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Number of terms which should be fixed and updated first. Defaults to <cite>None</cite>,
resulting in the number of unannotated factors being updated first.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A vector containing the initial update order of the terms</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="model-fitting">
<h2>Model fitting<a class="headerlink" href="#model-fitting" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="fscLVM.core.CSparseFA">
<em class="property">class </em><code class="descclassname">fscLVM.core.</code><code class="descname">CSparseFA</code><span class="sig-paren">(</span><em>init_data=None</em>, <em>E1=None</em>, <em>E2=None</em>, <em>**parameters</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA" title="Permalink to this definition">¶</a></dt>
<dd><p>Variational Bayesian Factor analysis module. <cite>AExpressionModule</cite> is definded in bayesnet.expressionnet</p>
<dl class="method">
<dt id="fscLVM.core.CSparseFA.getAnnotations">
<code class="descname">getAnnotations</code><span class="sig-paren">(</span><em>unannotated=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getAnnotations"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getAnnotations" title="Permalink to this definition">¶</a></dt>
<dd><p>Get annotations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>unannotated</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Booelan variable indicating whether unannotated annotations should be
generated by thresholding the prior at 0.5. Defaults to <cite>True</cite></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getDefaultParameters">
<code class="descname">getDefaultParameters</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getDefaultParameters"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getDefaultParameters" title="Permalink to this definition">¶</a></dt>
<dd><p>return a hash with default parameter value for this model</p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getF">
<code class="descname">getF</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getF"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getF" title="Permalink to this definition">¶</a></dt>
<dd><p>Get imputed expression values</p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getFactors">
<code class="descname">getFactors</code><span class="sig-paren">(</span><em>unannotated=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getFactors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getFactors" title="Permalink to this definition">¶</a></dt>
<dd><p>Get factors</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>unannotated</strong> &#8211; Boolean variable indicating whether to return also unannotated factors. DEfaults to true</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getName">
<code class="descname">getName</code><span class="sig-paren">(</span><em>base_name='fscLVM'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getName"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getName" title="Permalink to this definition">¶</a></dt>
<dd><p>return a name summarising the  main parameters</p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getNchanged">
<code class="descname">getNchanged</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getNchanged"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getNchanged" title="Permalink to this definition">¶</a></dt>
<dd><p>Return number of annotations changed by the model (sum of included and exluded genes )</p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getRelevance">
<code class="descname">getRelevance</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getRelevance"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getRelevance" title="Permalink to this definition">¶</a></dt>
<dd><p>Get posterior relevance (inverse of ARD score) <span class="math">\(1/Q(      extbf{lpha)}\)</span></p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getTerms">
<code class="descname">getTerms</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getTerms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getTerms" title="Permalink to this definition">¶</a></dt>
<dd><p>Get terms</p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getW">
<code class="descname">getW</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getW"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getW" title="Permalink to this definition">¶</a></dt>
<dd><p>Get weights (continous part of spike-and-slab prior) <span class="math">\(Q(\widetilde{W})\)</span></p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.getZ">
<code class="descname">getZ</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.getZ"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.getZ" title="Permalink to this definition">¶</a></dt>
<dd><p>Get posterior of Z (Bernourlli part part of spike-and-slab prior) <span class="math">\(Q(Z)\)</span></p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.iterate">
<code class="descname">iterate</code><span class="sig-paren">(</span><em>nIterations=None</em>, <em>forceIterations=False</em>, <em>tolerance=1e-08</em>, <em>minIterations=700</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.iterate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.iterate" title="Permalink to this definition">¶</a></dt>
<dd><p>Iterate updates of weights (with spike-and-slab prior), ARD parameters, factors, annd noise parameters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>nIternation</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Number of iterations.</li>
<li><strong>forceIterations</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Force the model to update <cite>nIteration</cite> times.</li>
<li><strong>tolerance</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#float" title="(in Python v2.7)"><em>float</em></a>) &#8211; Tolerance to monitor convergence of reconstruction error</li>
<li><strong>minIterations</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Minimum number of iterations the model should perform.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.printDiagnostics">
<code class="descname">printDiagnostics</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.printDiagnostics"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.printDiagnostics" title="Permalink to this definition">¶</a></dt>
<dd><p>Print diagnostics of the model. If more than 100% of gene annotations are for at least one factor, the model should be re-fitted with sparse unannotated facotrs.</p>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.regressOut">
<code class="descname">regressOut</code><span class="sig-paren">(</span><em>idx</em>, <em>use_latent=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.regressOut"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.regressOut" title="Permalink to this definition">¶</a></dt>
<dd><p>Regress out unwanted variation</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>idx</strong> (<em>vector_like</em>) &#8211; Indices of factors to be regressed out</li>
<li><strong>use_latent</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Boolean varoable indicating whether to regress out
the unwanted variation on the low-dimensional latent
space or the high-dimensional gene expression space.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A matrix containing the corrected expression values.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="fscLVM.core.CSparseFA.update">
<code class="descname">update</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/core.html#CSparseFA.update"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.core.CSparseFA.update" title="Permalink to this definition">¶</a></dt>
<dd><p>Perform update of weights (with spike-and-slab prior), ARD parameters, factors, annd noise parameters. Called by <cite>iterate</cite> method.</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="plotting-and-saving-results">
<h2>Plotting and saving results<a class="headerlink" href="#plotting-and-saving-results" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="fscLVM.plotTerms">
<code class="descclassname">fscLVM.</code><code class="descname">plotTerms</code><span class="sig-paren">(</span><em>FA=None</em>, <em>S=None</em>, <em>alpha=None</em>, <em>terms=None</em>, <em>madFilter=0.4</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#plotTerms"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.plotTerms" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot terms and their respective relevance</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>FA</strong> (<code class="xref py py-class docutils literal"><span class="pre">fscLVM.CSparseFA</span></code>) &#8211; Factor analysis object, ususally generated using <cite>initFA</cite> function</li>
<li><strong>madFilter</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#float" title="(in Python v2.7)"><em>float</em></a>) &#8211; Filter factors by this mean absolute deviation to exclude outliers.
For large datsets this can be set to 0.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fscLVM.plotFactors">
<code class="descclassname">fscLVM.</code><code class="descname">plotFactors</code><span class="sig-paren">(</span><em>FA</em>, <em>idx1=0</em>, <em>idx2=1</em>, <em>X=None</em>, <em>lab=None</em>, <em>terms=None</em>, <em>cols=None</em>, <em>isCont=True</em>, <em>madFilter=0.4</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#plotFactors"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.plotFactors" title="Permalink to this definition">¶</a></dt>
<dd><p>Scatter plot of 2 factors</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>FA</strong> (<code class="xref py py-class docutils literal"><span class="pre">fscLVM.CSparseFA</span></code>) &#8211; Factor analysis object, ususally generated using <cite>initFA</cite> function</li>
<li><strong>idx1</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Index of first factor to be plotted</li>
<li><strong>idx2</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Index of second factor to be plotted</li>
<li><strong>lab</strong> (<em>vector_like</em>) &#8211; Vector of labels for each data point</li>
<li><strong>isCont</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Boolean variable indicating whether labels should be interpreted as discrete or continuous</li>
<li><strong>cols</strong> (<em>vector_like</em>) &#8211; Vector of colors. Should be the same length as unique labels. Default is <cite>None</cite>,
then the <cite>brewer2mpl</cite></li>
<li><strong>madFilter</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#float" title="(in Python v2.7)"><em>float</em></a>) &#8211; Filter factors by this mean absolute deviation to exclude outliers.
For large datsets this can be set to 0.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fscLVM.plotFA">
<code class="descclassname">fscLVM.</code><code class="descname">plotFA</code><span class="sig-paren">(</span><em>FA</em>, <em>Nactive=20</em>, <em>stacked=True</em>, <em>db='MSigDB'</em>, <em>madFilter=0.4</em>, <em>unannotated=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#plotFA"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.plotFA" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot results of f-scLVM</p>
<p>Identified factors and corresponding gene set size ordered by relevance (white = low relevance; black = high relevance).
Top panel: Gene set augmentation, showing the number of genes added (red) and removed (blue) by the model for each factor.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>FA</strong> (<code class="xref py py-class docutils literal"><span class="pre">fscLVM.CSparseFA</span></code>) &#8211; Factor analysis object, ususally generated using <cite>initFA</cite> function</li>
<li><strong>Nactive</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#int" title="(in Python v2.7)"><em>int</em></a>) &#8211; Numer of terms to be plotted</li>
<li><strong>stacked</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Boolean variable indicating whether bars should be stacked</li>
<li><strong>db</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; Name of database used, eihter &#8216;MSigDB&#8217; or &#8216;REACTOME&#8217;</li>
<li><strong>madFilter</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#float" title="(in Python v2.7)"><em>float</em></a>) &#8211; Filter factors by this mean absolute deviation to exclude outliers.
For large datsets this can be set to 0.</li>
<li><strong>unannotated</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Indicates whether also unannotated factors should be plotted. Defaults to False.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fscLVM.saveFA">
<code class="descclassname">fscLVM.</code><code class="descname">saveFA</code><span class="sig-paren">(</span><em>FA</em>, <em>out_name=None</em>, <em>saveF=False</em>, <em>out_dir='./'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#saveFA"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.saveFA" title="Permalink to this definition">¶</a></dt>
<dd><p>Saves output of fscLVM.CSparseFA object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>FA</strong> (<code class="xref py py-class docutils literal"><span class="pre">fscLVM.CSparseFA</span></code>) &#8211; Factor analysis object, ususally generated using <cite>initFA</cite> function</li>
<li><strong>out_name</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; Name of hdf file to save the model to. Default is <a href="#id1"><span class="problematic" id="id2">`</span></a>None&#8217; for which a filename is created automatically.</li>
<li><strong>saveF</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#bool" title="(in Python v2.7)"><em>bool</em></a>) &#8211; Boolean variable indicating whether to save the imputed expression space.</li>
<li><strong>out_dir</strong> (<a class="reference external" href="https://docs.python.org/2/library/functions.html#str" title="(in Python v2.7)"><em>str</em></a>) &#8211; data directory where to save the outpout</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fscLVM.dumpFA">
<code class="descclassname">fscLVM.</code><code class="descname">dumpFA</code><span class="sig-paren">(</span><em>FA</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/fscLVM/utils.html#dumpFA"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#fscLVM.dumpFA" title="Permalink to this definition">¶</a></dt>
<dd><p>Dumps output of fscLVM.CSparseFA object in a dictionary</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>FA</strong> (<code class="xref py py-class docutils literal"><span class="pre">fscLVM.CSparseFA</span></code>) &#8211; Factor analysis object, ususally generated using <cite>initFA</cite> function</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">dictionary with results</td>
</tr>
</tbody>
</table>
</dd></dl>

<p>Contents:</p>
<div class="toctree-wrapper compound">
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="#">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">fscLVM documentation</a><ul>
<li><a class="reference internal" href="#tutorial">Tutorial</a><ul>
<li><a class="reference internal" href="#input">Input</a></li>
<li><a class="reference internal" href="#model-initialisation-and-fitting">Model initialisation and fitting</a></li>
<li><a class="reference internal" href="#diagnostics-plotting-and-saving">Diagnostics, plotting and saving.</a></li>
</ul>
</li>
<li><a class="reference internal" href="#loading-data-and-model-initialisation">Loading data and model initialisation</a></li>
<li><a class="reference internal" href="#model-fitting">Model fitting</a></li>
<li><a class="reference internal" href="#plotting-and-saving-results">Plotting and saving results</a></li>
</ul>
</li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/index.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="nav-item nav-item-0"><a href="#">fscLVM 1.0.0.dev1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2016, Florian Buettner and Oliver Stegle.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.8.
    </div>
  </body>
</html>